### PR TITLE
Add method to the Collection class to determine the number of items per year

### PR DESCRIPTION
**Pull request by SantiSite**

### CODE REVIEW

File: tests/entities/test_collection.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import unittest
+
+from bibx._entities.article import Article
+from bibx._entities.collection import Collection
+
+
+class TestCollection(unittest.TestCase):
```
Comment (odarbelaeze) [inline tests/entities/test_collection.py:7]: You can skip the `unittest.TestCase` here and run the tests using `pytest`. I think the command is `hatch run pytest`.

Comment (SantiSite) [reply tests/entities/test_collection.py:7]: Ok :+1:

File: tests/entities/test_collection.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
+from bibx._entities.article import Article
+from bibx._entities.collection import Collection
+
+
+class TestCollection:
```
Comment (odarbelaeze) [inline tests/entities/test_collection.py:5]: The `class` can also be omited, just the function called `test_something` is enough.

Comment (SantiSite) [reply tests/entities/test_collection.py]: :+1:

File: src/bibx/_entities/collection.py [hunk 3]
```diff
@@ -38,6 +39,27 @@ def citation_pairs(self) -> Iterable[Tuple[Article, Article]]:
...
+        :return: a dictionary with the number of articles published each year.
+        """
+        current_year = datetime.date.today().year
+        primary_year = min([article.year for article in self.articles])
+        years = {}
+        for year in range(primary_year, current_year + 1):
+            years[year] = 0
+
+        for article in self.articles:
+            if article.year in years:
```
Comment (odarbelaeze) [inline src/bibx/_entities/collection.py:57]: We need to handle the case where `article.year` is `None`. If that's the case we just ignore the article.

Comment (SantiSite) [reply src/bibx/_entities/collection.py:59]: :+1:

**Review by dani-vh — APPROVED** [review]

**Pull request merged**